src/ipc/ipc-common/IPCProxy.h

```diff
@@ -61,45 +61,48 @@ class IPCProxy : public WrapperType, public IPCProxyNewBase
     {
         QObject::connect(ipc(), &IPCProxyBinderBase::complete, this, [this] () {
                 m_localProviderBinder.init();
-                if (m_localProviderBinder.provider() == nullptr) {
-                    auto serviceName = ipc()->serviceName();
-                    auto objPath = ipc()->objectPath();
-                    for (auto& proxy : m_ipcProxies) {
-                        auto proxyAdapterIPCBinder = proxy.ipcBinder;
-                        if (proxyAdapterIPCBinder != nullptr) {
-                            proxyAdapterIPCBinder->setObjectPath(objPath);
-                            if(!serviceName.isEmpty()) {
-                                proxyAdapterIPCBinder->setServiceName(serviceName);
-                            }
-                            proxyAdapterIPCBinder->connectToServer();
-                            QObject::connect(proxyAdapterIPCBinder, &IPCProxyBinderBase::serviceAvailableChanged, this, &IPCProxy::refreshProvider);
-                        }
+                for (auto& proxy : m_ipcProxies) {
+                    auto ipcBinder = proxy.ipcBinder;
+                    if (ipcBinder != nullptr) {
+                        QObject::connect(ipcBinder, &IPCProxyBinderBase::serviceAvailableChanged, this, &IPCProxy::refreshProvider);
                     }
                 }
                 this->refreshProvider();
-            });
+        });
+    }
 
+    void enableBinder(IPCProxyBinderBase * ipcBinder)
+    {
+        ipcBinder->setObjectPath(ipc()->objectPath());
```

Comment:
It will call connectToServer succesfully if isReadyToConnect returns true. Then our connectToServer is not needed below. IMO this hidden (and early) connection should be somehow disabled.

Reply:
Alternatively add relevant comment here